addons/web/static/src/scss/bs_mixins_overrides.scss

```diff
@@ -51,6 +51,12 @@ $o-color-extras-nesting-selector: '&' !default;
         $-yiq-threshold-met: alpha($color) > $yiq-min-opacity-threshold;
 
         $-yiq-color: if($text-color, $text-color, if($-yiq-threshold-met, color-contrast($color, $main-background: $background), null));
+
+        // Generate utility classes for bg-* classes to then use them within
+        // the .badge design definition (see website.scss)
+        --bg-light: #{$color};
+        --color-light: #{$-yiq-color};
+
```

Comment:
@chgo-odoo 
I wonder if the fix might just be adding
```scss
    @each $-color, $-value in $grays {
        &.bg-#{$-color} {
            --bg-light: #{$-value};
            --color-light: #{color-contrast($-value, $min-contrast-ratio: 3)};
        }
    }
 ```
 to the `.badge, .o_filter_tag {` rules in `website.scss`
 right next to what is done with `$theme-colors` for theme colors.
 What do you think @qsm-odoo ?

Reply:
@chgo-odoo Indeed, but it feels like something is wrong here with using variables, because we cannot apply the styles already present.
![image](https://github.com/user-attachments/assets/13a45450-f6e3-4899-a2f1-7989dc7226bf)
I prefer that Quentin checks this, as he sees more clearly than me in all these CSS rules. ;)